src/commands/new/file.ts

```diff
@@ -160,11 +160,16 @@ export default class NewFile extends Command {
         console.log(`File ${fileNameToWriteToDisk} already exists. Ignoring...`);
         return;
       }
-    } catch (e) {
-      // File does not exist. Proceed creating it...
+    } catch (e:any ) {
+      
+      if (e.code === 'EACCES') {
+        console.error('Permission denied to read the file. You do not have the necessary permissions.');
+      } else {
+        await writeFile(fileNameToWriteToDisk, asyncApiFile, { encoding: 'utf8' });
```

Comment:
I would leave the "creation of the file" outside the `catch`, so it's not be treated as an error like I explain [here](https://github.com/asyncapi/cli/pull/812#issuecomment-1730136889).

Reply:
ok @peter-rr will do that.

Reply:
@peter-rr should i create file even though there is the permission error? means even though we dont have permission to read the file we just created it. it will look like this. 
![image](https://github.com/asyncapi/cli/assets/78146753/ff9c1084-b917-43cf-a85a-0b32ba5933b5)

Reply:
@Souvikns

Reply:
till then i have ran the lint command please consider merging it.

Reply:
> @peter-rr should i create file even though there is the permission error? means even though we dont have permission to read the file we just created it. it will look like this.

Nope, we should not create the file if a permission error occurs. In that case we should throw the error so that we exit the execution and the file is not created then:
```
} catch (e) {
      if (e.code === 'EACCES') {
        throw e;
      }
}
await writeFile(fileNameToWriteToDisk, asyncApiFile, { encoding: 'utf8' });
console.log(`Created file ${fileNameToWriteToDisk}...`);
```
 Another alternative we have is using `this.error` instead, since OCLIF makes use of it as `console.error` + `throw e`:
```
} catch (e) {
      if (e.code === 'EACCES') {
        this.error('Permission denied to read the file. You do not have the necessary permissions.');
      }
}
await writeFile(fileNameToWriteToDisk, asyncApiFile, { encoding: 'utf8' });
console.log(`Created file ${fileNameToWriteToDisk}...`);
```

Reply:
ok will do that.